r/R/singlecell_utils.R

```diff
@@ -68,4 +68,102 @@ marker_features <- function(mat, groups, method="wilcoxon") {
         foreground_mean = as.numeric(foreground_means),
         background_mean = as.numeric(background_means)
     )
-}
\ No newline at end of file
+}
+
+#' Aggregate counts matrices by cell group or feature.
+#' 
+#' Given a `(features x cells)` matrix, group cells by `cell_groups` and aggregate counts by `method` for each
+#' feature.
+#' @param cell_groups (Character/factor vector)  of cell groups/clusters. Length must be equal to the number of cells.
+#' @param method (Character vector) Method(s) to aggregate counts. If one method is provided, the output will be a matrix. If multiple methods are provided, the output will be a named list of matrices.
+#' 
+#' Current options are:
+#'  - `nonzeros`
+#'  - `sum`
+#'  - `mean`
+#'  - `var`
+#' @param clip_values (integer) If set, clip matrix values to the `<clip_values>`th quantile of each cell, between 0 and 1 inclusive.
```

Comment:
Two notes:

- I left clip_values here, but I'm still open to changing it into a completely different function.  I left it here so I could test it together with `pseudobulk_matrix()` a little easier.  Any thoughts?
- If we do leave it here, we should probably change the name of the arg.  Any idea for something more intuitive?